samples/cpp/dbt_face_detection.cpp

```diff
@@ -1,4 +1,4 @@
-#if defined(__linux__) || defined(LINUX) || defined(__APPLE__) || defined(ANDROID)
+#if defined(__linux__) || defined(LINUX) || defined(__APPLE__) || defined(ANDROID) || (defined(_MSC_VER) && _MSC_VER>1700)
```

Comment:
I tried to dig this out ( https://msdn.microsoft.com/en-us/library/hh567368.aspx#concurrencytable ) but can't find anything certain. TBH, the code was running on 2013 with minor modifications, but on 2015 it was throwing system_error according to standard. 
I don't have 2012 available so cannot confirm either compilation nor functionality.